examples/md-flexible/src/domainDecomposition/DomainTools.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file DomainTools.cpp
+ * @author J. Körner
+ * @date 13.05.2021
+ */
+#include "DomainTools.h"
+
+#include <cmath>
+#include <list>
+
+namespace DomainTools {
+bool isInsideDomain(const std::vector<double> &coordinates, std::vector<double> &boxMin, std::vector<double> &boxMax) {
+  bool isInsideLocalDomain = true;
+  for (int i = 0; i < coordinates.size(); ++i) {
+    if (!isInsideLocalDomain) {
+      break;
+    }
+    isInsideLocalDomain = coordinates[i] >= boxMin[i] && coordinates[i] < boxMax[i];
+  }
+  return isInsideLocalDomain;
+}
+
+bool isInsideDomain(const std::array<double, 3> &coordinates, std::array<double, 3> &boxMin,
+                    std::array<double, 3> &boxMax) {
+  bool isInsideLocalDomain = true;
+  for (int i = 0; i < coordinates.size(); ++i) {
+    if (!isInsideLocalDomain) {
+      break;
+    }
+    isInsideLocalDomain = coordinates[i] >= boxMin[i] && coordinates[i] < boxMax[i];
+  }
+  return isInsideLocalDomain;
+}
+
+double getDistanceToDomain(const std::vector<double> &coordinates, std::vector<double> &boxMin,
+                           std::vector<double> &boxMax) {
+  if (coordinates.size() == boxMin.size() && coordinates.size() == boxMax.size()) {
+    std::vector<double> differences(coordinates.size(), 0.0);
+    for (int i = 0; i < coordinates.size(); ++i) {
+      if (coordinates[i] < boxMin[i]) {
+        differences[i] = boxMin[i] - coordinates[i];
+      } else if (coordinates[i] > boxMax[i]) {
+        differences[i] = coordinates[i] - boxMax[i];
+      }
+    }
+
+    double distance = 0.0;
+    for (const auto &difference : differences) {
+      distance += std::pow(difference, 2.0);
+    }
+
+    return std::pow(distance, 1.0 / differences.size());
+  }
+  return -1;
+}
+
+double getDistanceToDomain(const std::array<double, 3> &coordinates, std::array<double, 3> &boxMin,
+                           std::array<double, 3> &boxMax) {
+  if (coordinates.size() == boxMin.size() && coordinates.size() == boxMax.size()) {
+    std::vector<double> differences(coordinates.size(), 0.0);
+    for (int i = 0; i < coordinates.size(); ++i) {
+      if (coordinates[i] < boxMin[i]) {
+        differences[i] = boxMin[i] - coordinates[i];
+      } else if (coordinates[i] > boxMax[i]) {
+        differences[i] = coordinates[i] - boxMax[i];
```

Comment:
`std::abs()`?

Reply:
This logic cannot be represented by std::abs. In the first case we use boxMin[i] and in the second case boxMax[i]. So even if we would use std::abs we still would neet the two branches.

Reply:
`std::clamp()` might also be useful here. Something like: `differences[i] = std::abs(coordinates[i] - std::clamp(coordinates[i], boxMin[i], boxMax[i]));`.  `std::abs()` is theoretically not even necessary here, since the value is squared later anyways.

Reply:
Totally missed that min max was mixed here, my bad. `std::clamp` looks better.